### ISSUE TITLE
Maint: doc build error, `dev` docs not updating

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#8432]
It looks like there is a fatal error in the docbuild, most likely pandas incompatibility

seems to be the same in main after each of the last few merges
doc build in some PRs was still green (?)

`dev` docs were last updated Aug.27

```
WARNING: 
>>>-------------------------------------------------------------------------
Warning in /home/runner/work/statsmodels/statsmodels/docs/source/contrasts.rst at block ending on line 166
Specify :okwarning: as an option in the ipython:: block to suppress this message
----------------------------------------------------------------------------
<ipython-input-41-a29d75116d4c>:1: FutureWarning: The behavior of `series[i:j]` with an integer-dtype index is deprecated. In a future version, this will be treated as *label-based* indexing, consistent with e.g. `series[i]` lookups. To retain the old behavior, use `series.iloc[i:j]`. To get the future behavior, use `series.loc[i:j]`.
  grouped.mean()["write"][4] - grouped.mean()["write"][:3].mean()
<<<-------------------------------------------------------------------------

Exception occurred:
  File "/opt/hostedtoolcache/Python/3.10.7/x64/lib/python3.10/site-packages/IPython/sphinxext/ipython_directive.py", line 600, in process_input
    raise RuntimeError('Non Expected warning in `{}` line {}'.format(filename, lineno))
RuntimeError: Non Expected warning in `/home/runner/work/statsmodels/statsmodels/docs/source/contrasts.rst` line 166
The full traceback has been saved in /tmp/sphinx-err-tiuz6r1l.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:39: html] Error 2
Error: Process completed with exit code 2.
```

**Comment by josef-pkt** [comment]
@bashtage Do you have any idea where to start debugging?

I don't know what has changed that might have caused this.

**Comment by josef-pkt** [comment]
docbuild started to fail when pandas 1.5.0 was used instead of pandas 1.4.3

first failing merge is for #8411, that's the first time pandas 1.5.0 was used
https://github.com/statsmodels/statsmodels/actions/runs/3086160158/jobs/4990229240#step:11:2156

Collecting pandas>=1.0
  Downloading pandas-1.5.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.1 MB)

**Comment by josef-pkt** [comment]
I guess the underlying problem is the pandas incompatibility in the future warning, i.e. not using `iloc`.
However, a problem like this should not be fatal, so there is still something fishy.

I will prepare a PR to change the constrast notebook

**Comment by bashtage** [comment]
The ipython blocks in the rst files assume no warnings unless explicitly
allowed.  I'll fix later.

**Comment by josef-pkt** [comment]
Now that we (or I) know where to look for problems, it might be better to keep that the docbuild breaks.
Otherwise we will not easily find out whether there are problems with the rst docs.

**Comment by josef-pkt** [comment]
fixed in #8433
